### PR TITLE
fix(docs): fix typescript errors

### PR DESCRIPTION
**Pull request by raichev-dima**
Should fix some typescript errors in docs

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/docs/components/docs-content/DocsContent.vue [hunk 1]
```diff
@@ -22,6 +21,6 @@ import { ApiDocsBlock } from '../../types/configTypes'
   },
 })
 export default class DocsContent extends Vue {
-  @Prop() readonly config: ApiDocsBlock[]
+  @Prop() readonly config?: ApiDocsBlock[]
```
Comment (asvae) [inline packages/docs/components/docs-content/DocsContent.vue:24]: ? -> !

Comment (asvae) [reply packages/docs/components/docs-content/DocsContent.vue:24]: Also required: true

File: packages/docs/components/dynamic-template/DynamicTemplate.vue [hunk 2]
```diff
@@ -17,39 +17,45 @@
...
-    return {
-      blockTags: {
-        [BlockType.TITLE]: 'h1',
-        [BlockType.SUBTITLE]: 'h3',
-        [BlockType.PARAGRAPH]: 'p',
-        [BlockType.CODE]: 'pre',
-        [BlockType.HEADLINE]: 'h5',
-      },
-    }
+  private blockTags: Record<string, any> = {
```
Comment (asvae) [inline packages/docs/components/dynamic-template/DynamicTemplate.vue:25]: `Record<BlockType, string>` ?

File: packages/docs/components/dynamic-template/DynamicTemplate.vue [hunk 3]
```diff
@@ -17,39 +17,45 @@
...
+    [BlockType.PARAGRAPH]: 'p',
+    [BlockType.CODE]: 'pre',
+    [BlockType.HEADLINE]: 'h5',
   }
 
+  @Prop({ required: true }) readonly block!: ApiDocsBlock
+
   get isCode () {
-    return this.block.type === BlockType.CODE
+    return this.block?.type === BlockType.CODE
```
Comment (asvae) [inline packages/docs/components/dynamic-template/DynamicTemplate.vue:36]: ? shouldn't be needed

File: packages/docs/components/dynamic-template/DynamicTemplate.vue [hunk 4]
```diff
@@ -17,39 +17,45 @@
...
   }
 
   get textToKebab () {
-    return kebabCase(this.$t(this.block.text))
+    if (!this.block) {
```
Comment (asvae) [inline packages/docs/components/dynamic-template/DynamicTemplate.vue:40]: Same

File: packages/docs/components/dynamic-template/DynamicTemplate.vue [hunk 5]
```diff
@@ -17,39 +17,45 @@
...
   }
 
   get isSubtitle () {
-    return this.block.type === BlockType.SUBTITLE
+    return this.block?.type === BlockType.SUBTITLE
   }
 
   get tag () {
-    return this.block.type === BlockType.COMPONENT ? this.block.component : this.blockTags[this.block.type]
+    if (!this.block) {
```
Comment (asvae) [inline packages/docs/components/dynamic-template/DynamicTemplate.vue:52]: same

File: packages/docs/types/configTypes.ts [hunk 6]
```diff
@@ -15,18 +16,26 @@ export enum BlockType {
...
-  type: BlockType.TITLE | BlockType.SUBTITLE | BlockType.PARAGRAPH | BlockType.HEADLINE,
-  translationString: TranslationString,
-}
+      type:
+        | BlockType.TITLE
+        | BlockType.SUBTITLE
+        | BlockType.PARAGRAPH
+        | BlockType.HEADLINE,
+      translationString: TranslationString,
+      text: string,
```
Comment (asvae) [inline packages/docs/types/configTypes.ts:25]: This is incorrect. There should be no text.
In any of examples.

File: packages/docs/components/dynamic-template/DynamicTemplate.vue (hunk 2 above)
Comment (raichev-dima) [reply packages/docs/components/dynamic-template/DynamicTemplate.vue:25]: @asvae `Partial<Record<BlockType, string>>`?

File: packages/docs/types/configTypes.ts (hunk 6 above)
Comment (raichev-dima) [reply packages/docs/types/configTypes.ts:25]: @asvae How about this:
image

Comment (asvae) [reply packages/docs/types/configTypes.ts:25]: Translation string should be fine here. It's used for anchors I suppose?

File: packages/docs/components/docs-content/DocsContent.vue [hunk 7]
```diff
@@ -1,18 +1,17 @@
 <template>
-  <va-content>
-    <dynamic-template
-      v-for="(block, index) in config"
-      :key="index"
-      :block="block"
-    />
-  </va-content>
+<va-content>
```
Comment (asvae) [inline packages/docs/components/docs-content/DocsContent.vue:2]: lint

File: packages/docs/components/dynamic-template/DynamicTemplate.vue [hunk 8]
```diff
@@ -1,55 +1,50 @@
...
+  <component v-else :is="blockTags[block.type]">
+    {{ $t(block.translationString) }}
   </component>
 </template>
 
 <script lang="ts">
 import { Component, Vue, Prop } from 'vue-property-decorator'
-import { ApiDocsBlock, BlockType } from '../../types/configTypes.ts'
+import { ApiDocsBlock, BlockType, TextBlock } from '../../types/configTypes'
+import Example from '../../components/Example.vue'
```
Comment (asvae) [inline packages/docs/components/dynamic-template/DynamicTemplate.vue:22]: 2 words at least. ApiExample or something.

**Pull request merged**